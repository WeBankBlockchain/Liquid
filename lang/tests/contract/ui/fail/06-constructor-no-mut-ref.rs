use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&self) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
