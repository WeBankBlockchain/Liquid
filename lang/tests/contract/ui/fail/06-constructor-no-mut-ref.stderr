error: `&mut self` is mandatory first parameter for constructor of contract
  --> $DIR/06-constructor-no-mut-ref.rs:10:20
   |
10 |         pub fn new(&self) {}
   |                    ^^^^^
